presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -249,23 +248,7 @@ public long getTotalBytes()
     @Override
     public long getCompletedBytes()
     {
-        if (!closed) {
-            updateCompletedBytes();
-        }
-        return completedBytes;
-    }
-
-    private void updateCompletedBytes()
-    {
-        try {
-            long newCompletedBytes = (long) (totalBytes * recordReader.getProgress());
-            completedBytes = min(totalBytes, max(completedBytes, newCompletedBytes));
-        }
-        catch (IOException ignored) {
-        }
-        catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-        }
+        return dataSource.getReadBytes();
```

Comment:
I don't think you should change the behavior of the record cursor because the code only uses dataSource to read the footer and the stats, and instead uses a direct HDFS input stream for reading the data.